name: Test

on:
  push:
    branches:
      - main
  pull_request:

jobs:
  actionlint:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    permissions:
      checks: write
      contents: read
      pull-requests: write
    steps:
      - uses: actions/checkout@v5
      - name: Run actionlint with reviewdog
        uses: reviewdog/action-actionlint@f00ad0691526c10be4021a91b2510f0a769b14d0 # v1.68.0
        with:
          fail_level: error
          filter_mode: nofilter
          level: error
          reporter: github-pr-review
  codeql:
    permissions:
      actions: read
      checks: read
      contents: read
      security-events: write
    uses: route06/actions/.github/workflows/codeql.yml@3bcea5bf834ff2e0e6c912c1167fcc1fe05e49c2 # v2.7.0
  pushover:
    name: pushover if failure
    if: github.ref_name == github.event.repository.default_branch && failure()
    permissions: {}
    needs: [actionlint, codeql]
    uses: ./.github/workflows/pushover.yml
    secrets:
      PUSHOVER_API_KEY: ${{ secrets.PUSHOVER_API_KEY }}
      PUSHOVER_USER_KEY: ${{ secrets.PUSHOVER_USER_KEY }}
